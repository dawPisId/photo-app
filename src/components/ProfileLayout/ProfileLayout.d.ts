export interface ProfileLayoutProps {
  user: {
    name: string;
    phone: string;
    username: string;
    website: string;
    email: string;
    address: {
      street: string;
      city: string;
      suite: string;
      zipcode: string;
    };
    company: {
      name: string;
      catchPhrase: string;
      bs: string;
    };
  };
}
