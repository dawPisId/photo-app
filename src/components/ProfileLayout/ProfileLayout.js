import {
  CompanyLayout,
  FirstHeader,
  Label,
  Section,
  SectionHeader,
  SectionLayout
} from "./ProfileLayout.styled";

import InputField from "../InputField/InputField";
import PermContactCalendarIcon from "@material-ui/icons/PermContactCalendar";
import React from "react";

function ProfileLayout({ personalInfo, addressInfo, companyInfo }) {
  let personal = [
    { name: "Name", value: personalInfo.name },
    { name: "Phone", value: personalInfo.phone },
    { name: "Username", value: personalInfo.username },
    { name: "Website", value: personalInfo.website },
    { name: "E-mail", value: personalInfo.email },
  ];
  let address = [
    { name: "Street", value: addressInfo.street },
    { name: "City", value: addressInfo.city },
    { name: "Sufix", value: addressInfo.suite },
    { name: "Zipcode", value: addressInfo.zipcode },
  ];

  let company = [
    { name: "Name", value: companyInfo.name },
    { name: "Catchphrase", value: companyInfo.catchPhrase },
    { name: "Bs", value: companyInfo.bs },
  ];
  return (
    <div>
      <Section>
        <FirstHeader>
          <PermContactCalendarIcon />
          <Label>Personal information</Label>
        </FirstHeader>
        <SectionLayout>
          {personal.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </SectionLayout>
        <SectionHeader>
          <PermContactCalendarIcon />
          <Label>Address</Label>
        </SectionHeader>
        <SectionLayout>
          {address.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </SectionLayout>
        <SectionHeader>
          <PermContactCalendarIcon />
          <Label>Company</Label>
        </SectionHeader>
        <CompanyLayout>
          {company.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </CompanyLayout>
      </Section>
    </div>
  );
}

export default ProfileLayout;
