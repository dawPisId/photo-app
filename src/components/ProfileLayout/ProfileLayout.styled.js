import styled from "styled-components";

export const Section = styled.div``;
export const Label = styled.div`
  font-family: "Roboto", sans-serif;
  font-size: 1.5rem;
`;

export const FirstHeader = styled.div`
  display: flex;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid lightgrey;
  svg {
    padding-right: 1rem;
  }
`;

export const SectionHeader = styled.div`
  display: flex;
  align-items: center;
  padding: 1rem;
  border-top: 1px solid lightgrey;
  border-bottom: 1px solid lightgrey;
  svg {
    padding-right: 1rem;
  }
`;

export const SectionLayout = styled.div`
  display: grid;
  padding: 3rem 2rem 3rem 2rem;
  grid-template-columns: 1.05fr 0.95fr;
  grid-gap: 1rem;
  grid-column-gap: 4rem;
  @media(max-width:730px){
      grid-template-columns: 1fr;
  }
`;

export const CompanyLayout = styled.div`
  display: flex;
  flex-direction: column;
  padding: 3rem 2rem 3rem 2rem;
  gap:1rem;
`;
