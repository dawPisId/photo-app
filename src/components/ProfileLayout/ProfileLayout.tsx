import {
  CompanyLayout,
  FirstHeader,
  Label,
  Section,
  SectionHeader,
  SectionLayout,
} from "./ProfileLayout.styled";
import React, { FC } from "react";

import InputField from "../InputField/InputField";
import PermContactCalendarIcon from "@material-ui/icons/PermContactCalendar";
import { ProfileLayoutProps } from "./ProfileLayout.d";

const ProfileLayout: FC<ProfileLayoutProps> = ({ user }) => {
  const personal = [
    { name: "Name", value: user.name },
    { name: "Phone", value: user.phone },
    { name: "Username", value: user.username },
    { name: "Website", value: user.website },
    { name: "E-mail", value: user.email },
  ];
  const address = [
    { name: "Street", value: user.address.street },
    { name: "City", value: user.address.city },
    { name: "Sufix", value: user.address.suite },
    { name: "Zipcode", value: user.address.zipcode },
  ];

  const company = [
    { name: "Name", value: user.company.name },
    { name: "Catchphrase", value: user.company.catchPhrase },
    { name: "Bs", value: user.company.bs },
  ];
  return (
    <div>
      <Section>
        <FirstHeader>
          <PermContactCalendarIcon />
          <Label>Personal information</Label>
        </FirstHeader>
        <SectionLayout>
          {personal.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </SectionLayout>
        <SectionHeader>
          <PermContactCalendarIcon />
          <Label>Address</Label>
        </SectionHeader>
        <SectionLayout>
          {address.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </SectionLayout>
        <SectionHeader>
          <PermContactCalendarIcon />
          <Label>Company</Label>
        </SectionHeader>
        <CompanyLayout>
          {company.map(({ name, value }, index) => (
            <InputField label={name} fillValue={value} key={index} />
          ))}
        </CompanyLayout>
      </Section>
    </div>
  );
};

export default ProfileLayout;
