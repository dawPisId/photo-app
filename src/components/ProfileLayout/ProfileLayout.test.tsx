import ProfileLayout from "./ProfileLayout";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tempUser = {
    name: "",
    phone: "",
    username: "",
    website: "",
    email: "",
    address: {
      street: "",
      city: "",
      suite: "",
      zipcode: "",
    },
    company: {
      name: "",
      catchPhrase: "",
      bs: "",
    },
  };
  const tree = renderer.create(<ProfileLayout user={tempUser} />).toJSON();
  expect(tree).toMatchSnapshot();
});
