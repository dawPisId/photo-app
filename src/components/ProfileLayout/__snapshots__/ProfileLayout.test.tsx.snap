// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-bdnxRM"
  >
    <div
      className="sc-dlnjwi kUjdMX"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1z"
        />
      </svg>
      <div
        className="sc-gtsrHT eCbWQp"
      >
        Personal information
      </div>
    </div>
    <div
      className="sc-eCApnc hheWKK"
    >
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Name
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Phone
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Username
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Website
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          E-mail
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-hKFxyN dcfYAR"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1z"
        />
      </svg>
      <div
        className="sc-gtsrHT eCbWQp"
      >
        Address
      </div>
    </div>
    <div
      className="sc-eCApnc hheWKK"
    >
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Street
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          City
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Sufix
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Zipcode
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-hKFxyN dcfYAR"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1z"
        />
      </svg>
      <div
        className="sc-gtsrHT eCbWQp"
      >
        Company
      </div>
    </div>
    <div
      className="sc-jSFjdj euLUMp"
    >
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Name
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Catchphrase
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-gKAaRy cGdSQQ"
      >
        <span
          className="sc-fujyAs fByZWv"
        >
          Bs
        </span>
        <input
          className="sc-iCoGMd jbmZBi"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
